api/envoy/config/listener/v3/listener.proto

```diff
@@ -103,7 +103,6 @@ message Listener {
   }
 
   // Configuration for envoy internal listener. All the future internal listener features should be added here.
-  // [#not-implemented-hide:]
   message InternalListenerConfig {
```

Comment:
Also maybe worth to note, this will make the listener use the listener name as the internal address. And when this is internal listener, the address field should keep empty.